### ISSUE TITLE
suspendedTransactionAsync not returning connections to pool

### ISSUE THREAD

**Issue by ncobc** [JetBrains/Exposed#680]
Exposed 0.17.7
Kotlin 1.3.50

The number of queries I can make (with each returning a result before the next is initiated) is equal to the maximum pool size. Then, the process hangs and an exception gets thrown because no connections are available. Regular transactions work fine.

Edit: The suspending function is being called from a ktor "get" block which also uses coroutines. Is this a problem?

```kotlin
// Create data source.
internal val dataSource: HikariDataSource
    get() {
        val config = HikariConfig().apply {
            driverClassName = "org.postgresql.Driver"
            jdbcUrl = System.getenv("JDBC_DATABASE_URL")
            validate()
        }

        return HikariDataSource(config)
    }

// Connect using data source, create table if missing.
fun initdb() {
    Database.connect(dataSource)

    transaction { SchemaUtils.createMissingTablesAndColumns(Spaces) }
}

// Reduce boilerplate.
suspend fun <T> dbquery(block: () -> T): T = suspendedTransactionAsync(Dispatchers.IO) {
    block()
}.await()

// Query (can be executed maximumPoolSize times).
suspend fun readAll() = dbquery { Spaces.selectAll() }
```

**Comment by Tapac** [comment]
@ncobc , that's strange as I can see that `close` should be called on a connection and that means what connection should be returned to a pool.

Could you provide a sample project to test?

**Comment by sgerber-hyperanna** [comment]
We can replicate this, in the particular case of an exception being thrown from the suspended transaction.

**Comment by sgerber-hyperanna** [comment]
Actually... it seems to only affect 0.17.6.

I was creating a test project to replicate the issue and can confirm our problem is present in 0.17.6 and resolved in 0.17.7
